node/src/reactor/joiner.rs

```diff
@@ -213,15 +213,17 @@ impl From<StorageRequest> for Event {
 
 impl From<NetworkRequest<NodeId, Message>> for Event {
     fn from(request: NetworkRequest<NodeId, Message>) -> Self {
-        Event::SmallNetwork(small_network::Event::from(request))
+        Event::SmallNetwork(small_network::Event::NetworkRequest {
+            req: Box::new(request),
```

Comment:
As stated above, **if** we need to do this, do not instantiate these events manually, but add a `From` `impl`, I'd say.

But please, check what's "wrong" first, causing this to no longer be a newtype-like struct.

Reply:
Implemented specific `impl From`